File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Roadmap/16 - EXPRESIONES REGULARES/python/rikmij.py
import re

texto_ejemplo = "2 y dos son 4, cuatro y dos son 6, seis y dos son 8 y ocho 16"

'''find_nums = re.findall(r'\d+', texto_ejemplo)

print(find_nums)'''
find_nums = r'\d+'
for elem in re.finditer(find_nums, texto_ejemplo):
    first = elem.start()
    last = elem.end()
    print(f"{texto_ejemplo[first:last]} -> posiciones ({first}-{last})")

print('\n', '~'*7, "EJERCICIO EXTRA", '~'*7)
#Validar email
email = r'\w+\S*@\w+.\w+'
texto_email = "Buenas tardes, el correo electrónico es correo.deprueba_hola@corporacion.es. Muchas gracias"
print(re.search(email, texto_email))

#Validar número teléfono
telf = r'[+]\d+\s\d+'
texto_telf = "Puede llamar al número +34 6874521035200 para más información"
print(re.search(telf,texto_telf))

#Validar URL
web = r'https?://\w+.\w+.\w+[.\w+]*'
texto_web = "Puede visitar nuestra web: https://www.unaweb_deprueba101.com.wordpress.es para más información"
print(re.search(web, texto_web))

File: Roadmap/16 - EXPRESIONES REGULARES/kotlin/rikmij.kt
fun main() {
    val textoEjemplo = "2 y dos son 4, cuatro y dos son 6, seis y dos son 8 y ocho 16"
    val findNums = Regex("\\d+").findAll(textoEjemplo)
    for (num in findNums) {
        val first = num.range.first
        val last = num.range.last
        println("${num.value} -> ($first,$last)")
    }

    println("${"\n" + "~".repeat(7)} EJERCICIO EXTRA ${"~".repeat(7)}")
    //Validar email
    val email = Regex("\\w+\\S*@\\w+.\\w+")
    val textEmail = "Buenas tardes, el correo electrónico es correo.deprueba_hola@corporacion.es. Muchas gracias"
    val findEmail = email.find(textEmail)
    println(findEmail?.value)

    //Validar teléfono
    val telf = Regex("[+]\\d+\\s\\d+")
    val textTelf = "Puede llamar al número +34 6874521035200 para más información"
    val findTelf = telf.find(textTelf)
    println(findTelf?.value)

    //Validar URL
    val url = Regex("https?://\\w+.\\w+.\\w+[.\\w+]*")
    val textUrl = "Puede visitar nuestra web: https://www.unaweb_deprueba101.com.wordpress.es para más información"
    val findUrl = url.find(textUrl)
    println(findUrl?.value)
}
